### PR TITLE
Chapter 10 Exercise 08

### PR DESCRIPTION
**Pull request by archiebrowne**

### CODE REVIEW

File: M1FExplained/Chapter10/Exercises/Exercise08.lean [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+namespace Chapter10.Exercise08
+
+/-
+Let n ≥ 2 be an integer. Prove that n is prime if and only if for every integer a, 
+either gcd(a, n) = 1 or n ∣ a. 
+-/
+
+lemma exercise08 (n : ℤ) (hn : 2 ≤ n) : Prime n ↔ ∀ (a : ℤ), Int.gcd a n = 1 ∨ n ∣ a := by
+  -- Since 2 ≤ n, it is a natural number, we can rewrite this in the local context.
+  rw [show n = n.natAbs by
```
Comment (kbuzzard) [inline M1FExplained/Chapter10/Exercises/Exercise08.lean]: Using `Int.natAbs_of_nonneg` would probably be quicker here

File: M1FExplained/Chapter10/Exercises/Exercise08.lean [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    apply Or.elim (Classical.em (N ∣ a.natAbs))
+    <;> intro h'
+  -- If N ∣ |a|, then trivially N ∣ |a|.
+    · right
+      exact Iff.mpr Int.ofNat_dvd_left h'
+  -- If ¬ N ∣ |a|, then since N is prime, gcd(N, a) = 1. This is proven by contradiction.
+    · left
+      by_contra h''
+      set M := Int.gcd a N with hM
+  -- Condition on wether a = 0 or not.
```
Comment (kbuzzard) [inline M1FExplained/Chapter10/Exercises/Exercise08.lean]: You never use condition `ha` in the "not zero" case, so you can just not do this case split and use the second argument to do the a=0 case as well.

File: M1FExplained/Chapter10/Exercises/Exercise08.lean [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      apply Or.elim (Classical.em (a = 0))
+      <;> intro ha
+  -- If a = 0, then N ∣ a, which contradicts h'
+      · rw [ha] at h'
+        simp at h'
+  -- If a ≠ 0, then M = gcd(a, N) divides N, and so is equal to 1 or N, both of which are contradictions.
```
Comment (kbuzzard) [inline M1FExplained/Chapter10/Exercises/Exercise08.lean]: This is also true if a=0

**Review by kbuzzard — COMMENTED** [review]
Looks good to me -- thanks!

**Comment by archiebrowne** [comment]
changes pushed

**Comment by kbuzzard** [comment]
Thanks!

**Pull request merged**